Source/Peripheral.swift

```diff
@@ -123,7 +123,7 @@ public class Peripheral {
      Immediately after that `.Complete` is emitted.
      */
     public func discoverServices(_ serviceUUIDs: [CBUUID]?) -> Observable<[Service]> {
-        if let identifiers = serviceUUIDs, let services = self.services?.filter({ identifiers.contains($0.uuid) }), identifiers.count == services.count {
+        if let identifiers = serviceUUIDs, let services = self.services?.filter({ identifiers.contains($0.uuid) }), identifiers.count <= services.count {
             return ensureValidPeripheralState(for: .just(services))
         }
         let observable = peripheral.rx_didDiscoverServices
```

Comment:
From what I can see `self.peripheral.discoverServices(serviceUUIDs)` is always called, even when cache is sufficient for our needs.